CoreLib/SharedPtr.h

```diff
@@ -0,0 +1,490 @@
+            ++(_control_block->_weak_count);
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline WeakPtr<T, Deleter>::WeakPtr(WeakPtr<T, Deleter>&& other) noexcept
+        : _control_block(std::exchange(other._control_block, nullptr))
+    {
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline WeakPtr<T, Deleter>& WeakPtr<T, Deleter>::operator=(const WeakPtr<T, Deleter>& other) noexcept
+    {
+        if (this == &other)
+            return *this;
+
+        Release();
+
+        _control_block = other._control_block;
+        ++(_control_block->_weak_cout);
+
+        return *this;
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline WeakPtr<T, Deleter>& WeakPtr<T, Deleter>::operator=(WeakPtr<T, Deleter>&& other) noexcept
+    {
+        if (this == &other)
+            return *this;
+
+        Release();
+
+        _control_block = std::exchange(other._control_block, nullptr);
+
+        return *this;
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline WeakPtr<T, Deleter>::~WeakPtr()
+    {
+        Release();
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline WeakPtr<T, Deleter>::WeakPtr(const SharedPtr<T, Deleter>& shared) noexcept
+        : _control_block(shared.GetControlBlock())
+    {
+        ++(_control_block->_weak_count);
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline WeakPtr<T, Deleter>& WeakPtr<T, Deleter>::operator=(const SharedPtr<T, Deleter>& shared) noexcept
+    {
+        Release();
+        _control_block = shared.GetControlBlock();
+        ++(_control_block->_weak_count);
+
+        return *this;
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline SharedPtr<T, Deleter> WeakPtr<T, Deleter>::Lock() const noexcept
+    {
+        if (_control_block && _control_block->_strong_count > 0)
+            return SharedPtr<T, Deleter>(_control_block);
+
+        return SharedPtr<T, Deleter>();
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline bool WeakPtr<T, Deleter>::Expired() const noexcept
+    {
+        return !_control_block || _control_block->_strong_count == 0;
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline void WeakPtr<T, Deleter>::Release() noexcept
+    {
+        if (_control_block)
+            --(_control_block->_weak_count);
+
+        _control_block = nullptr;
+    }
+
+    template<class T, class Deleter> requires IsDefaultCompatibleDeleter<T, Deleter>
+    inline void WeakPtr<T, Deleter>::Swap(WeakPtr<T, Deleter>& other) noexcept
+    {
+        std::swap(_control_block, other._control_block);
+    }
+
+
```

Comment:
Too much blank lines

Reply:
These are different blocks - division between blocks WeakPtr in SharedPtr

Reply:
It's better to mark it with a comment then